File: leetcode/5longestPalindromic/Solution.java
class Solution {

    public Boolean judgePalindrome(String s) {
        int length = s.length();
        for (int i = 0; i <= (length-i-1); i++) {
            if (s.charAt(i) != s.charAt(length - 1 - i)) {
                return false;
            }
        }
        return true;
    }

    public String longestPalindrome(String s) {
        String maxStr = "";
        for (int i = 0; i + maxStr.length() < s.length(); i++) {
            for (int j = s.length() - 1; j >= i + maxStr.length(); j--) {
                System.out.println(j);
                System.out.println("++++++++++++===");
                if (s.charAt(i) == s.charAt(j) && this.judgePalindrome(s.substring(i, j+1))) {
                    maxStr = s.substring(i, j+1);
                }
            }
        }
        return maxStr;
    }

    public static void main(String[] args) {
        String s1 = "cbbdbb";
        String s2 = "a";
        String s3 = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
        Solution testSolution = new Solution();
        // System.out.println(testSolution.longestPalindrome(s1));

        // System.out.println(testSolution.longestPalindrome(s2));
        System.out.println(testSolution.longestPalindrome(s3));
    }
}

File: leetcode/5longestPalindromic/solution.js
#!/usr/bin/env node

var judgePalindrome = function(s) {
    // return (s.split("").reverse().join("") == s);
    const len = s.length;
    for (let i = 0; i <= (len-i-1); i++) {
        if (s[i] !== s[len - i - 1]) {
            return false;
        }
    }
    return true;
};

/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {
    let maxStr = "";
    let maxLength = 0;
    let length = s.length;
    for (let i = 0; i + maxLength < length; i++) {
        for (let j = length-1; j >= i + maxLength; j--) {
            let tempStr = s.substr(i, j-i+1);
            if (s[i] == s[j] && judgePalindrome(tempStr)) {
                maxStr = tempStr;
                maxLength = j+1-i;
            }
        }
    }
    return maxStr;
};

let string1 = "cbbdbb";
let string2 = "a";
let string3 = "cbbd";
let string4 = "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa";
let string5 = "jglknendplocymmvwtoxvebkekzfdhykknufqdkntnqvgfbahsljkobhbxkvyictzkqjqydczuxjkgecdyhixdttxfqmgksrkyvopwprsgoszftuhawflzjyuyrujrxluhzjvbflxgcovilthvuihzttzithnsqbdxtafxrfrblulsakrahulwthhbjcslceewxfxtavljpimaqqlcbrdgtgjryjytgxljxtravwdlnrrauxplempnbfeusgtqzjtzshwieutxdytlrrqvyemlyzolhbkzhyfyttevqnfvmpqjngcnazmaagwihxrhmcibyfkccyrqwnzlzqeuenhwlzhbxqxerfifzncimwqsfatudjihtumrtjtggzleovihifxufvwqeimbxvzlxwcsknksogsbwwdlwulnetdysvsfkonggeedtshxqkgbhoscjgpiel"
let res = longestPalindrome(string5);
// let res = judgePalindrome(string3);
console.log("=============");
console.log(res);
